harvester/NexusRepository.py

```diff
@@ -0,0 +1,282 @@
+    """ Nexus Repository """
+
+    def setRepoParams(self, repoParams):
+        self.metadataprefix = "nexus"
+        super(NexusRepository, self).setRepoParams(repoParams)
+        self.domain_metadata = []
+        self.headers = {"accept": "application/ld+json"}
+
+    def _crawl(self):
+        kwargs = {
+            "repo_id": self.repository_id, "repo_url": self.url, "repo_set": self.set, "repo_name": self.name,
+            "repo_type": "nexus",
+            "enabled": self.enabled, "repo_thumbnail": self.thumbnail, "item_url_pattern": self.item_url_pattern,
+            "abort_after_numerrors": self.abort_after_numerrors,
+            "max_records_updated_per_run": self.max_records_updated_per_run,
+            "update_log_after_numitems": self.update_log_after_numitems,
+            "record_refresh_days": self.record_refresh_days,
+            "repo_refresh_days": self.repo_refresh_days, "homepage_url": self.homepage_url,
+            "repo_oai_name": self.repo_oai_name,
+            "repo_registry_uri": self.repo_registry_uri
+        }
+        self.repository_id = self.db.update_repo(**kwargs)
+
+        try:
+            item_count = 0
+            query_url = self.url
+            while query_url:
+                querystring = {"type": "https://schema.org/Dataset"}
+                response = requests.request("GET", query_url, headers=self.headers, params=querystring)
+                response = response.json()
+
+                for record in response["_results"]:
+                    item_identifier = record["_self"]
+                    self.db.write_header(item_identifier, self.item_url_pattern, self.repository_id)
+                    item_count = item_count + 1
+                    if (item_count % self.update_log_after_numitems == 0):
+                        tdelta = time.time() - self.tstart + 0.1
+                        self.logger.info("Done {} item headers after {} ({:.1f} items/sec)".format(item_count,
+                                                                                                   self.formatter.humanize(
+                                                                                                       tdelta),
+                                                                                                   item_count / tdelta))
+
+                if "_next" in response:
+                    query_url = response["_next"]
+                else:
+                    break
+            self.logger.info("Found {} items in feed".format(item_count))
+
+            return True
+
+        except Exception as e:
+            self.logger.error("Updating Nexus Repository failed: {} {}".format(type(e).__name__, e))
+            self.error_count = self.error_count + 1
+            if self.error_count < self.abort_after_numerrors:
+                return True
+
+        return False
+
+    def format_nexus_to_oai(self, nexus_dats_json):
+
+        nexus_record = nexus_dats_json
```

Comment:
Why the copy here?

Reply:
Because I was copying from https://github.com/frdr-dfdr/frdr_harvest/blob/master/harvester/DataStreamRepository.py#L67, where I wrote equally bad code :) will fix in both